.github/workflows/main.yml

```
@@ -1,50 +1,120 @@
+# Can be run locally using act and docker, or in Fedora 39 also with podman, using:
+# act --bind --container-daemon-socket $XDG_RUNTIME_DIR/podman/podman.sock -W .github/workflows/main.yml
+
 name: Unit tests
 
-on: [push, pull_request]
+on: pull_request
+env:
+  PYTHONWARNINGS: "ignore"
 
 jobs:
-  test_py2:
-    runs-on: ubuntu-20.04
-
+  test:
+    strategy:
+      max-parallel: 1
+      matrix:
+        include:
+        - python-version: '3.11'
+          os: ubuntu-latest
+        - python-version: '2.7'
+          os: ubuntu-20.04
+        - python-version: '3.10'
+          os: ubuntu-latest
+        - python-version: '3.9'
+          os: ubuntu-latest
+        - python-version: '3.8'
+          os: ubuntu-latest
+        - python-version: '3.7'
+          os: ubuntu-latest
+        - python-version: '3.6'
+          os: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    permissions:
+      # Gives the action the necessary permissions for publishing new
+      # comments in pull requests.
+      pull-requests: write
+      # Gives the action the necessary permissions for pushing data to the
+      # python-coverage-comment-action branch, and for editing existing
+      # comments (to avoid publishing multiple comments in the same PR)
+      contents: write
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
-          fetch-depth: 0
-      - name: Set up Python 2.7
-        uses: actions/setup-python@v2
+          # ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0  # To get changed lines for coverage-diff: origin/master..HEAD
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
         with:
-          python-version: '2.7'
+          python-version: ${{ matrix.python-version }}
 
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements-dev.txt
-          # FIXME: branding.py still has no permanent home
-          curl https://gist.github.com/ydirson/3c36a7e19d762cc529a6c82340894ccc/raw/5ca39f621b1feab813e171f535c1aad1bd483f1d/branding.py -O -L
-          pip install pyliblzma
-          pip install -e .
-          command -v xz
-
-      - name: Test
+          python -m pip install --quiet --upgrade pip
+          pip install --quiet -r requirements-dev.txt coveralls # FIXME:
+          if [ ${{ matrix.python-version }} = 2.7 ]; then python2 setup.py install --user;
+          else pip install --quiet -e .;fi
```

Why is the local package installed here? doesn't look like it should be part of the dependencies step

In any case, if it's needed I recommend having two gated jobs here:
```suggestion
      - name: Install python 2 dependencies
        if: ${{ matrix.python-version == 2.7 }}
        run: |
          python -m pip install --quiet --upgrade pip
          pip install --quiet -r requirements-dev.txt coveralls
          python2 setup.py install --user

      - name: Install python 3 dependencies
        if: ${{ matrix.python-version != 2.7 }}
        run: |
          python -m pip install --quiet --upgrade pip
          pip install --quiet -r requirements-dev.txt coveralls
          pip install --quiet -e .
```

O maybe separate the dependencies step to be a shared one and the install package into 2, one for pytohn 2 and one for python 3